mcapi/mcapi.proto

```diff
@@ -0,0 +1,437 @@
+      // the report start time. If the end date is not the same day of the week
+      // a partial week will be reported and the last value will conincide with
+      // the last day of the report.
+      WEEKLY = 2;
+      // Report metrics across the entire span from report start to report end.
+      TOTAL = 3;
+    }
+    // The frequency for this page of results.
+    Frequency frequency = 1;
+
+    // Whether the metrics should be cumulative. This option has no effect
+    // when the frequency is TOTAL.
+    bool cumulative = 2;
+
+    // A future option to support unique reach across time periods.
+    // TODO: This shoudl live in a message with cumulative
+    // bool unique_reach_by_time_period = 4;
+
+    Grouping grouping = 4;
+
+    // A conjunction of filters
+    repeated Filter filters = 5;
+  }
+  // A map of calculation specs that can be reused across
+  // Page Definitions.
+  map<string, CalculationSpec> calculation_spec_map = 4;
+
+  // The definition for a page of results
+  message PageDefinition {
+    // The title of the page
+    string title = 1;
+
+    // Specifies the calculation spec to be used by this Page
+    // Definition.
+    string calculation_spec_key = 2;
+
+    // If specified the total population for the specified groupings and
+    // filters that are associated with calculation spec is reported.
+    bool population_size = 3;
+
+    // The set of Metrics to be computed
+    message MetricSpec {
+      // Understanding this is VERY important.
+      //
+      // The resoure names of the Reporting Sets to report on.
+      //
+      // If all Reporting Sets are associated with a single media type,
+      // the impression qualification filter for that media type will be
+      // applied. Otherwise, if multiple media types are represented the cross
+      // media filter will be used.
+      //
+      // Recall that a single ImpressionQualificationFilter pertains to a
+      // single Media Type and that this is at most of such filter per Basic
+      // Report. Also recall that a single ImpressionQualificationFilter can
+      // contain multiple filters to be applied to that Media Type (e.g.
+      // AMI, MRC, Custom).
+      //
+      // The effect of this is that for each filter present in the
+      /// Impression Qualification Filter for the detected Media Type
+      // a distinct set of metrics is computed.
+      //
+      // For example if a reporting set named "RS" is specified and
+      // its Media Type is video then the system will find the
+      // Impression Qualification Filter (IQF) whose media_type is equal to
+      // VIDEO. Further assume that the VIDEO IQF has filter_names = [ami, mrc]
+      // and also specifies a custom filter. Then, in this case, a set of
+      // Metrics is generated for the ami, mrc, and custom filter. The specific
+      // types of metrics generated are defined by the MetricType structure
+      // below.
```

Comment:
We'll want to add a MediaType here and change the way reporting sets are interpreted, such that if they contain Event Groups of heterogeneous MediaType that the EGs are filtered to the MediaType specified here.

This would mean the API actually creates some additional RS objects for the homogeneous MediaType EventGroups.